### DISCUSSION TITLE
Whatever the status is, the default color on the badge is the same

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#6271] [Help]
Whatever the status is, the default color on the badge is the same.

```php
Tables\Columns\BadgeColumn::make('status')
    ->enum(OrderStatus::toOptions())
    ->colors([
        'danger'  => static fn ($state): bool => in_array($state, [OrderStatus::CANCELLED, OrderStatus::FAILED]),
        'warning' => OrderStatus::PROCESSING,
        'success' => OrderStatus::PAID,
    ]),
```
The state is paid
![7](https://user-images.githubusercontent.com/29732308/233485336-4cea8363-9876-496a-9ea1-b127fda04b55.png)

**Comment by VincvanAs** [comment]
Should you not call the state eg:
'succes' => static fn ($state): bool => $state === OrderStatus::PAID

**Reply by ArtMin96** [reply]
Thanks. I tried all options

**Comment by danharrin** [comment]
Is it a real enum or a class constant?

**Reply by ArtMin96** [reply]
It is a real enum
![7](https://user-images.githubusercontent.com/29732308/233626424-b238e6b6-aa72-40bf-b6b2-1fa886de3d96.png)

**Reply by ArtMin96** [reply]
enum:
```php
->enum([
    'processing' => 'Processing',
    'paid'       => 'Paid',
    'failed'     => 'Failed',
    'cancelled'  => 'Cancelled',
])
```

**Reply by ArtMin96** [reply]
Even if I delete `->enum()`, there is no change

**Reply by danharrin** [reply]
What is `$state` if you `dd()` it in the closure? Is it still an enum? Maybe enum comparisons for colors just don't work at the moment

**Reply by ArtMin96** [reply]
When I do `dd` it returns `paid`

**Reply by danharrin** [reply]
So somewhere in Filament the enum is not being passed through correctly and is transformed into a string

**Reply by ArtMin96** [reply]
There is the same problem without giving an enum

```php
Tables\Columns\BadgeColumn::make('status')
    ->colors([
        'danger'  => static fn ($state): bool => in_array($state, [OrderStatus::CANCELLED, OrderStatus::FAILED]),
        'warning' => OrderStatus::PROCESSING,
        'success' => OrderStatus::PAID,
    ]),
```

![7](https://user-images.githubusercontent.com/29732308/233647779-56b751a5-73ef-4fe7-b607-91dae75bebfd.png)

**Comment by adaoss** [comment]
Solved question using

                Tables\Columns\BadgeColumn::make('status')
                    ->enum(EntityStatus::getSelectOptions())
                    ->colors([
                        'default'  => EntityStatus::NEW->value,
                        'warning'  => EntityStatus::BLOCKED->value,
                        'warning'  => EntityStatus::INACTIVE->value,
                        'success' => EntityStatus::ACTIVE->value
                    ]),
